lib/ecmascript.ts

```diff
@@ -4526,7 +4529,7 @@ function DifferenceInstant(
   ns1: JSBI,
   ns2: JSBI,
   increment: number,
-  smallestUnit: keyof typeof nsPerTimeUnit,
+  smallestUnit: Parameters<typeof NS_PER_TIME_UNIT.get>[0],
```

Comment:
Minor thing, but IMO it'd make this more readable to use a custom type alias here, e.g. `TIME_OR_DAY_UNIT` or something like that.